examples/medplum-photon-integration/src/bots/create-photon-prescription.test.ts

```diff
@@ -0,0 +1,215 @@
+        coding: [
+          {
+            system: 'http://www.nlm.nih.gov/research/umls/rxnorm',
+            code: '310430',
+            display: 'gabapentin 100 MG Oral Capsule',
+          },
+        ],
+      },
+    });
+
+    await medplum.createResource({
+      resourceType: 'MedicationKnowledge',
+      code: {
+        coding: [
+          {
+            system: 'http://www.nlm.nih.gov/research/umls/rxnorm',
+            code: '310430',
+            display: 'gabapentin 100 MG Oral Capsule',
+          },
+          {
+            system: NEUTRON_HEALTH_TREATMENTS,
+            code: 'photon-treatment-id',
+          },
+        ],
+      },
+    });
+
+    const result = await handler(medplum, {
+      bot,
+      contentType,
+      secrets,
+      input: medicationRequest,
+    });
+
+    const photonPrescriptionId = result.identifier?.find((id) => id.system === NEUTRON_HEALTH);
+    expect(photonPrescriptionId).toBeDefined();
+  }, 10000);
+
+  test('Throw error if no medication code', async () => {
+    const medplum = new MockClient();
+    const medicationRequest: MedicationRequest = await medplum.createResource({
+      resourceType: 'MedicationRequest',
+      status: 'active',
+      intent: 'filler-order',
+      subject: { reference: 'Patient/123' },
+    });
+    await expect(() => handler(medplum, { bot, contentType, secrets, input: medicationRequest })).rejects.toThrow(
+      'Medication must have a code'
+    );
+  });
+});
+
+describe('Validate variables', () => {
+  test.skip('Create and validate variables', () => {
+    const result = createAndValidateVariables('patient-id', 'treatment-id', 30, 'tablets', 'Take one daily');
+    expect(result).toEqual({
+      patientId: 'patient-id',
+      treatmentId: 'treatment-id',
+      dispenseQuantity: 30,
+      dispenseUnit: 'tablets',
+      instructions: 'Take one daily',
+    });
+  });
+
+  test.skip('Error if missing variables', () => {
+    expect(() => createAndValidateVariables('patient-id', 'treatment-id', 30, 'tablets', undefined)).toThrow(
+      'The following required fields are missing: instructions'
+    );
+  });
+});
+
+describe('getPhotonTreatmentId', () => {
+  const mockAuthToken = 'mock-auth-token';
+
+  test('Get treatment ID from MedicationKnowledge', async () => {
+    const medplum = new MockClient();
+    const mockMedicationCode: CodeableConcept = {
+      coding: [{ system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456', display: 'Test Medication' }],
+    };
+
+    await medplum.createResource({
+      resourceType: 'MedicationKnowledge',
+      code: {
+        coding: [
+          { system: NEUTRON_HEALTH_TREATMENTS, code: 'photon-treatment-id' },
+          { system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456' },
+        ],
+      },
+    });
+
+    const result = await getPhotonTreatmentId(mockAuthToken, medplum, mockMedicationCode);
+    expect(result).toBe('photon-treatment-id');
```

Comment:
Test is failing as searching for the MedicationKnowledge by code is returning undefined.